### PR TITLE
Bind API for attributes and basic thread operations

### PR DESCRIPTION
**Pull request by mboes**

### CODE REVIEW

**Review by facundominguez — APPROVED** [review]
Don't you feel like moving to inline-c now that there are more bindings?

File: src/System/Posix/Thread.hsc [hunk 1]
```diff
@@ -1,4 +1,6 @@
 -- | Bindings to the POSIX threads library.
+--
+-- Requires linking with the @-threaded@ RTS.
```
Comment (facundominguez) [inline src/System/Posix/Thread.hsc:3]: The error isn't very good when the user forgets about this.
```
spec: schedule: re-entered unsafely.
   Perhaps a 'foreign import unsafe' should be 'safe'?
```

A runtime check in `create` would help.

**Comment by mboes** [comment]
> Don't you feel like moving to inline-c now that there are more bindings?

For sure. Didn't know whether you wanted to keep the dependency set to just `base` so I went along with the current binding style. But arguably `inline-c` might be more convenient.

File: src/System/Posix/Thread.hsc (hunk 1 above)
Comment (mboes) [reply src/System/Posix/Thread.hsc:3]: True. Will add. But I don't know of a better way to test for `-threaded` than to use `rtsSupportsBoundThreads`.

**Pull request merged**